internal/k8s/reconciler/gateway.go

```diff
@@ -20,15 +26,15 @@ type K8sGateway struct {
 	*gwv1beta1.Gateway
 	GatewayState *state.GatewayState
 
-	config apigwv1alpha1.GatewayClassConfig
+	Config apigwv1alpha1.GatewayClassConfig
```

Comment:
Exported this so that the JSON marshaler includes it when we serialize for writing to the store backend.

If not exported, we don't have any idea what the max connections value is when rehydrating from the store and syncing into Consul, which we do at an interval in the background for resiliency.